tests/integration/test_arrowflight_interface/test.py

```diff
@@ -0,0 +1,204 @@
+        writer, _ = client.do_put(descriptor, schema)
+        writer.write_batch(batch)
+        writer.close()
+        assert False, "Expected error but query succeeded"
+    except flight.FlightServerError:
+        pass
+
+
+def test_doget():
+    node.query(
+        """
+        CREATE TABLE doget_test (
+            id Int64,
+            name String
+        )
+        ORDER BY id
+        """
+    )
+    node.query(
+        """
+            INSERT INTO doget_test VALUES(10, 'abc'), (20, 'cde')
+        """)
+    
+    client = flight.FlightClient(f"grpc://{node.ip_address}:8888")
+
+    try:
+        incorrect_ticket = flight.Ticket(b"INSERT INTO doget_test_insert SELECT number, toString(number) FROM numbers(10)")
+        _ = client.do_get(incorrect_ticket)
+        assert False, "Expected error but query succeeded"
+    except flight.FlightServerError:
+        pass
+    
+    try:
+        incorrect_ticket = flight.Ticket(b"some incorrect query")
+        _ = client.do_get(incorrect_ticket)
+        assert False, "Expected error but query succeeded"
+    except flight.FlightServerError:
+        pass
+
+    try:
+        incorrect_ticket = flight.Ticket(b"SELECT * FROM unexisting_table")
+        _ = client.do_get(incorrect_ticket)
+        assert False, "Expected error but query succeeded"
+    except flight.FlightServerError:
+        pass
+    
+    real_ticket = flight.Ticket(b"SELECT * FROM doget_test")
+    reader = client.do_get(real_ticket)
+    actual = reader.read_all()
+
+    expected_schema = pa.schema([
+        pa.field("id",   pa.int64(),  nullable=False),
+        pa.field("name", pa.string(), nullable=False),
+    ])
+    expected_ids   = pa.chunked_array([pa.array([10, 20],          type=pa.int64())])
+    expected_names = pa.chunked_array([pa.array(["abc", "cde"],    type=pa.string())])
+
+    assert actual.schema == expected_schema
+
+    assert actual.column("id").equals(expected_ids)
+    assert actual.column("name").equals(expected_names)
+
+
+    real_ticket = flight.Ticket(b"SELECT id FROM doget_test")
+    reader = client.do_get(real_ticket)
+    actual = reader.read_all()
+
+    expected_schema = pa.schema([
+        pa.field("id",   pa.int64(),  nullable=False),
+    ])
+    expected_ids   = pa.chunked_array([pa.array([10, 20],          type=pa.int64())])
+
+    assert actual.schema == expected_schema
+
+    assert actual.column("id").equals(expected_ids)
+
+    node.query("DROP TABLE IF EXISTS doget_test SYNC")
+
+
+def test_doget_format_json():
+    node.query(
+        """
+        CREATE TABLE doget_test_format_json (
+            id Int64,
+            name String
+        )
+        ORDER BY id
+        """
+    )
+    node.query(
+        """
+            INSERT INTO doget_test_format_json VALUES(10, 'abc'), (20, 'cde')
+        """)
+    
+    client = flight.FlightClient(f"grpc://{node.ip_address}:8888")
+
+    real_ticket = flight.Ticket(b"SELECT * FROM doget_test_format_json FORMAT JSON")
```

Comment:
Please add some comments to this test about what we're testing here. For other tests it can be also useful.

Reply:
Probably it's better to throw an exception if ClickHouse gets via ArrowFlight protocol a query with explicit format and this format is not `Arrow`. Because such queries are kind of invalid.

Reply:
can we do this without regex?
Via AST. I do not know how to do this

Reply:
fixed